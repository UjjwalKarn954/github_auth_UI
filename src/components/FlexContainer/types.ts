export type Props = {
    gap?: string;
    minWidth?: string;
    flexWrap?: boolean;
    flexGrow?: boolean;
    alignItems?: 'flex-start' | 'flex-end' | 'center' | 'initial';
    justifyContent?: 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around' | 'space-evenly' | 'initial';
};
