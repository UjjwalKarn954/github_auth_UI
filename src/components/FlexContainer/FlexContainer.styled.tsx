import { styled } from '@styled';
import { Props } from './types';

export const FlexContainer = styled('div')<Props>`
    display: flex;
    padding: 1rem;
    gap: ${({ gap }) => gap};
    flex: ${({ flexGrow }) => (flexGrow ? 1 : 'auto')};
    flex-wrap: ${({ flexWrap }) => (flexWrap ? 'wrap' : 'nowrap')};
    min-width: ${({ minWidth }) => minWidth};
    align-items: ${({ alignItems }) => alignItems};
    justify-content: ${({ justifyContent }) => justifyContent};
`;

FlexContainer.defaultProps = {
    gap: '1rem',
    flexGrow: false,
    flexWrap: false,
    minWidth: 'auto',
    alignItems: 'initial',
    justifyContent: 'initial'
};
