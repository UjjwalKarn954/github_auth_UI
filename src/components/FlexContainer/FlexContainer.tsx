import React from 'react';
import * as Styled from './FlexContainer.styled';
import { Props } from './types';

export const FlexContainer: React.FC<Props> = React.memo(props => {
    return <Styled.FlexContainer {...props}>{props.children}</Styled.FlexContainer>;
});

FlexContainer.displayName = 'FlexContainer';
