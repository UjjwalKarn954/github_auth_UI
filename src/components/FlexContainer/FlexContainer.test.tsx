import { render, screen } from '@test-utils';
import React from 'react';
import { FlexContainer } from './FlexContainer';

describe('Flex Container', () => {
    it('should render properly with children', () => {
        const { container } = render(<FlexContainer>Sample text</FlexContainer>);
        expect(screen.getByText(/Sample text/i)).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it('should render properly with flex-grow and flex-wrap', () => {
        const { container } = render(
            <FlexContainer flexGrow flexWrap>
                Sample text
            </FlexContainer>
        );
        expect(container).toMatchSnapshot();
    });
});
