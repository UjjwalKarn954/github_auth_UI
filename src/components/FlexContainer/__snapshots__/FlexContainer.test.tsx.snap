// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Container should render properly with children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2rem;
  position: fixed;
  z-index: 1001;
  top: 0;
  right: 0;
}

.c0 > * + * {
  margin-top: 2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  gap: 1rem;
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  min-width: auto;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
}

<div>
  <div
    class="c0"
  />
  <div
    class="c1"
  >
    Sample text
  </div>
</div>
`;

exports[`Flex Container should render properly with flex-grow and flex-wrap 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2rem;
  position: fixed;
  z-index: 1001;
  top: 0;
  right: 0;
}

.c0 > * + * {
  margin-top: 2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  gap: 1rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  min-width: auto;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
}

<div>
  <div
    class="c0"
  />
  <div
    class="c1"
  >
    Sample text
  </div>
</div>
`;
